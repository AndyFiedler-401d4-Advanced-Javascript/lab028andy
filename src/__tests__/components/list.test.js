import React from 'react';
import renderer from 'react-test-renderer';
import List from '../../components/list';

describe('<List />', () => {
  it('can render initially in ul', () => {
    let app = shallow(<List items={[]}/>);
    expect(app.find('ul').exists()).toBe(true);
    expect(app.find('ul li').exists()).toBe(false);
  });

  it('can add todos by passing props', () => {

    let app = mount(<List items={[{id: 0, text: 'test', completed: true }]} />);
    expect(app.find('ul li').exists()).toBe(true);
  });

  it('renders consistently', () => {
    const tree = renderer.create(<List items={[]} />).toJSON();
    // console.log(tree);
    expect(tree).toMatchSnapshot();
  });
});
