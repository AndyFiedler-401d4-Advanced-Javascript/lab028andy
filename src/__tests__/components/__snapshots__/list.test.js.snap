// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders consistently 1`] = `
<ul
  id="items"
/>
`;
